control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -423,165 +423,153 @@ func SetupGatewayControllerWithManager(ctx context.Context, mgr ctrl.Manager, co
 			// Subscribe to changes from Consul for TCPRoutes
 			&source.Channel{Source: c.Subscribe(ctx, api.TCPRoute, r.transformConsulTCPRoute(ctx)).Events()},
 			&handler.EnqueueRequestForObject{},
 		).
-		Watches(
+		WatchesRawSource(
 			// Subscribe to changes from Consul for InlineCertificates
 			&source.Channel{Source: c.Subscribe(ctx, api.InlineCertificate, r.transformConsulInlineCertificate(ctx)).Events()},
 			&handler.EnqueueRequestForObject{},
 		).
-		Watches(
+		WatchesRawSource(
 			&source.Channel{Source: c.Subscribe(ctx, api.JWTProvider, r.transformConsulJWTProvider(ctx)).Events()},
 			&handler.EnqueueRequestForObject{},
 		).
 		Watches(
-			source.NewKindWithCache((&v1alpha1.GatewayPolicy{}), mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformGatewayPolicy(ctx)),
+			&v1alpha1.GatewayPolicy{},
+			handler.EnqueueRequestsFromMapFunc(r.transformGatewayPolicy),
 		).
 		Watches(
-			source.NewKindWithCache((&v1alpha1.RouteRetryFilter{}), mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformRouteRetryFilter(ctx)),
+			&v1alpha1.RouteRetryFilter{},
+			handler.EnqueueRequestsFromMapFunc(r.transformRouteRetryFilter),
 		).
 		Watches(
-			source.NewKindWithCache((&v1alpha1.RouteTimeoutFilter{}), mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformRouteTimeoutFilter(ctx)),
+			&v1alpha1.RouteTimeoutFilter{},
+			handler.EnqueueRequestsFromMapFunc(r.transformRouteTimeoutFilter),
 		).
 		Watches(
 			// Subscribe to changes in RouteAuthFilter custom resources referenced by HTTPRoutes.
-			source.NewKindWithCache((&v1alpha1.RouteAuthFilter{}), mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformRouteAuthFilter(ctx)),
+			&v1alpha1.RouteAuthFilter{},
+			handler.EnqueueRequestsFromMapFunc(r.transformRouteAuthFilter),
 		).
 		Complete(r)
 }
 
 // transformGatewayClass will check the list of GatewayClass objects for a matching
 // class, then return a list of reconcile Requests for it.
-func (r *GatewayController) transformGatewayClass(ctx context.Context) func(o client.Object) []reconcile.Request {
-	return func(o client.Object) []reconcile.Request {
-		gatewayClass := o.(*gwv1beta1.GatewayClass)
-		gatewayList := &gwv1beta1.GatewayList{}
-		if err := r.Client.List(ctx, gatewayList, &client.ListOptions{
-			FieldSelector: fields.OneTermEqualSelector(Gateway_GatewayClassIndex, gatewayClass.Name),
-		}); err != nil {
-			return nil
-		}
-		return common.ObjectsToReconcileRequests(pointersOf(gatewayList.Items))
+func (r *GatewayController) transformGatewayClass(ctx context.Context, o client.Object) []reconcile.Request {
+	gatewayClass := o.(*gwv1beta1.GatewayClass)
+	gatewayList := &gwv1beta1.GatewayList{}
+	if err := r.Client.List(ctx, gatewayList, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(Gateway_GatewayClassIndex, gatewayClass.Name),
+	}); err != nil {
+		return nil
 	}
+	return common.ObjectsToReconcileRequests(pointersOf(gatewayList.Items))
 }
```

Comment:
See above comment. We no longer need to create a closure here to consume our context since the context is provided to us at runtime now